### PR TITLE
Migrate Bugsnag handling

### PR DESCRIPTION
**Pull request by eborden**
This commit includes a variety of Bugsnag supports for freckle apps.

- Parsing env variables.
- Refined grouping of common exceptions (http, sql).
- Standard `beforeNotify` hooks to add refinements.

These have been extracted from Freckle's applications with domain
specific logic removed.

### CODE REVIEW

File: stack.yaml [hunk 1]
```diff
@@ -4,6 +4,8 @@ extra-deps:
   - hspec-junit-formatter-1.1.0.2
   - memcache-0.3.0.1
   - scientist-0.0.0.0
+  - bugsnag-1.0.0.1
+  - bugsnag-wai-1.0.0.1
```
Comment (eborden) [inline stack.yaml:8]: Wild guess that I'll need to update the other `stack.yaml` files :smile:

Comment (pbrisbin) [reply stack.yaml:8]: All but nightly ;) I just got this into Stackage recently so it should show up in the LTS soon too.

File: library/Freckle/App/Bugsnag.hs [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+  }
+
+-- | Set StackFrame's InProject to @'False'@ for Error Helper modules
+--
+-- We want exceptions grouped by the the first stack-frame that is /not/ them.
+-- Marking them as not in-project does this, with little downside.
+--
+maskErrorHelpers :: BeforeNotify
+maskErrorHelpers = setStackFramesInProjectByFile (`notElem` errorHelpers)
+  where errorHelpers = ["library/FrontRow/Core/Exceptions.hs"]
```
Comment (pbrisbin) [inline library/Freckle/App/Bugsnag.hs:145]: This is annoying, and I'm open to the practicality argument, but `freckle-app` really shouldn't know about this list of error helpers from our actual domain-specific app.

**Review by pbrisbin — APPROVED** [review]
Excellent. I've been meaning to do this!

File: library/Freckle/App/Bugsnag.hs (hunk 2 above)
Comment (eborden) [reply library/Freckle/App/Bugsnag.hs:145]: Yeah, totally missed that little line.

Comment (eborden) [reply library/Freckle/App/Bugsnag.hs:145]: One option is to have it check for `Exceptions` modules in general.

**Comment by pbrisbin** [comment]
Oh, and don't forget to update all the lock files...

```console
for x in stack-lts-*.yaml; do stack --stack-yaml $x build --dry-run; done
```

Yeah, this stuff's supes annoying.

**Review by mjgpy3 — APPROVED** [review]
LGTM -- nice tests!

File: library/Freckle/App/Bugsnag.hs (hunk 2 above)
Comment (eborden) [reply library/Freckle/App/Bugsnag.hs]: For now I'm just going to codify this convention. If we have a need then we can make it more configurable.

**Pull request merged**